src/components/__tests__/codeBlock.test.js

```diff
@@ -1,13 +1,35 @@
 import React from 'react';
 import {create} from 'react-test-renderer';
 
+import { CodeBlock } from '../codeBlock';
 import {_setCachedCodeKeywords, CodeContextProvider, DEFAULTS} from '../codeContext';
 import {makeKeywordsClickable} from '../codeKeywords';
 
 function CodeWrapper({children, ...props}) {
   return <code {...props}>{children ? makeKeywordsClickable(children) : children}</code>;
 }
 
+// Tests that CodeBlock renders SignInNote when necessary.
+describe('CodeBlock', () => {
+  beforeEach(() => {
+    _setCachedCodeKeywords(DEFAULTS);
+  });
+
+  it('renders a SignInNote above the code block if the code has a placeholder', () => {
+    const tree = create(
+      <CodeBlock filename="test" language="javascript">
+        process.env.MY_ENV = ___ORG_SLUG___
+      </CodeBlock>
+    );
+
+    expect(tree.toJSON()).toMatchSnapshot();
+  });
+
+  it('renders without SignInNote if the code does not contain keyword settings', () => {
+    <CodeBlock filename="test" language="javascript" />;
```

Comment:
I also think this code block needs to have content, otherwise it won't be rendered correctly